src/main/org/codehaus/groovy/classgen/AsmClassGenerator.java

```diff
@@ -1218,7 +1218,7 @@ public void visitVariableExpression(VariableExpression expression) {
 
         BytecodeVariable variable = controller.getCompileStack().getVariable(variableName, false);
         if (variable == null) {
-            processClassVariable(variableName);
+            processClassVariable(expression);
```

Comment:
Thanks for the PR.  I merged before seeing the comment from @blackdrag, so another refactor PR would be good.